### DISCUSSION TITLE
pnpm workspace + turbo + vercel: only redeploy works

### DISCUSSION THREAD

**Discussion by rsodre** [vercel/turborepo#9642] [Help] [answered]
### Summary

Hey!

I turned my typescript project into a monorepo, with pnpm workspaces. It's now a [client](https://github.com/underware-gg/pistols/tree/main/client) (next.js app) and an [sdk](https://github.com/underware-gg/pistols/tree/main/sdk) (library). The sdk is a dependency on the client.

Then I discovered that to make it deployable in Vercel, I had to use turbo, or else the client would not build the monorepo dependency. I never used it before, it solves a bunch of problems, loving it! Everything was fine on the work branch, and preview deployments were good.

Then I merged into main.
Every time I deploy, the client does not find the sdk.
Then I redeploy, and it works.
Every time.
What's wrong?

<img width="1199" alt="Screenshot 2024-12-20 at 17 47 36" src="https://github.com/user-attachments/assets/aa069ceb-9c72-4100-abd1-a7fed8f77800" />

<img width="811" alt="Screenshot 2024-12-20 at 17 46 29" src="https://github.com/user-attachments/assets/15c7c3ab-6c15-4171-8bfe-f84e0d743d2c" />

### Additional information

```js
* Followed this guide to setup tubo into my monorepo:
https://turbo.build/repo/docs/getting-started/add-to-existing-repository
* Followed this guide to setup turbo in Vercel:
https://vercel.com/docs/monorepos/turborepo
* Not using Remote Caching
* Vercel build command: `turbo run build`
```

### Example

https://github.com/underware-gg/pistols/tree/main/client

**Comment by anthonyshew** [comment, marked as answer]
Hi, welcome to Turborepo!

The reason you're seeing this is that the `dist` directory for `sdk` is not being cached. Turborepo only knows to cache directories and files that are in [the `outputs` key](https://turbo.build/repo/docs/reference/configuration#outputs), so a cache hit won't restore `dist` given your `turbo.json`.

Luckily, the fix is simple: Add a `"dist/**"` member to your `outputs` array, and then the outputs for the `sdk` will be cached for the times when you hit cache.

**Reply by rsodre** [reply]
after that change, first deployment worked!
thank you!
